src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/updateindexmetadata/UpdateManagedIndexMetaDataRequest.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.action.updateindexmetadata
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.ManagedIndexMetaData
+import org.elasticsearch.action.ActionRequestValidationException
+import org.elasticsearch.action.ValidateActions.addValidationError
+import org.elasticsearch.action.support.master.AcknowledgedRequest
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.index.Index
+
+class UpdateManagedIndexMetaDataRequest : AcknowledgedRequest<UpdateManagedIndexMetaDataRequest> {
+    var index: Index? = null
+        private set
+    var managedIndexMetaData: ManagedIndexMetaData? = null
+        private set
+
+    constructor()
+
+    constructor(index: Index, managedIndexMetaData: ManagedIndexMetaData) {
+        this.index = index
+        this.managedIndexMetaData = managedIndexMetaData
+    }
+
+    override fun validate(): ActionRequestValidationException? {
+        var validationException: ActionRequestValidationException? = null
+        if (index == null) {
+            validationException = addValidationError("must specify index for UpdateManagedIndexMetaData", validationException)
+        } else if (managedIndexMetaData == null) {
+            validationException = addValidationError("must specify managedIndexMetaData for UpdateManagedIndexMetaData", validationException)
+        }
+        return validationException
+    }
+
+    override fun writeTo(streamOutput: StreamOutput) {
+        super.writeTo(streamOutput)
+        index!!.writeTo(streamOutput)
```

Comment:
Try changing to lateinit so we don't have to use !! everywhere if we're sure the validate is always run before